### DISCUSSION TITLE
TLS handshake errors all of a sudden (alll nodes)

### DISCUSSION THREAD

**Discussion by prologic** [burmilla/os#122] [Q&A] [answered]
Anyone seen or getting this error on their BurmillaOS ndoes:

```
2021-11-19 20:42:39.136737 I | http: TLS handshake error from <refacted>:55554: remote error: tls: unknown certificate authority
```

This is occurring on all 6 nodes all of a sudden overnight. I didn't do anything here. Regenerating the certs also doesn't appear to help.

**Comment by olljanat** [comment]
Can you share more details about your configuration? You say all nodes so are running swarm, kubernetes, etc? Or are those single nodes?

Which OS version? Which hypervisor/bare metal?

Also where you see that error?

Probably it would good idea create issue and will details asked on it too.

**Reply by prologic** [reply]
I will answer and solve my own question.

TL;DR: Expired Root CA.

Turns out the following happened:

- My `docker-machine` CA expired.
- This is the CA used by all my Docker Swarm nodes running BurmillaOS (_I would never use Kubernetes :D_)
- Apparently `docker-machine regenerate-certs` **does not** care for, nor chec or attempt to update an old expired CA it used previously.
- But `docker-machine create -f -d generic ...` does.

Basically I manage all my (_still_) via `docker-machine`'s `generic` driver.

I _could_ switch to using the newer `docker context` over SSH tunnels, but I just can't be bothered changing my habits 😅

**Comment by prologic** [comment, marked as answer]
I will answer and solve my own question.

TL;DR: Expired Root CA.

Turns out the following happened:

- My `docker-machine` CA expired.
- This is the CA used by all my Docker Swarm nodes running BurmillaOS (_I would never use Kubernetes :D_)
- Apparently `docker-machine regenerate-certs` **does not** care for, nor chec or attempt to update an old expired CA it used previously.
- But `docker-machine create -f -d generic ...` does.

Basically I manage all my (_still_) via `docker-machine`'s `generic` driver.

I _could_ switch to using the newer `docker context` over SSH tunnels, but I just can't be bothered changing my habits 😅
